### PR TITLE
[CORE][VL][CH] Make Iceberg code implement component API

### PR DESCRIPTION
**Pull request by zhztheplayer**
Add `CHIcebergComponent` / `VeloxIcebergCompoent` to replace the previous injections through `ScanTransformerFactory`.

Component API was added in this PR: https://github.com/apache/incubator-gluten/pull/8143.

With other minor refactors.

### CODE REVIEW

File: gluten-iceberg/src/main/scala/org/apache/gluten/execution/OffloadIcebergScan.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    // Inject RAS rule.
+    injector.gluten.ras.injectRasRule {
+      c =>
+        RasOffload.Rule(
+          RasOffload.from[BatchScanExec](OffloadIcebergScan()),
+          Validators.newValidator(c.glutenConf),
+          Nil)
+    }
```
Comment (zhztheplayer) [inline gluten-iceberg/src/main/scala/org/apache/gluten/execution/OffloadIcebergScan.scala:55]: The code can be simplified in future PRs.

**Comment by zhztheplayer** [comment]
cc @liujiayi771 @zzcclp

**Review by PHILO-HE — APPROVED** [review]

File: gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala [hunk 2]
```diff
@@ -94,4 +98,8 @@ object IcebergScanTransformer {
       commonPartitionValues = SparkShimLoader.getSparkShims.getCommonPartitionValues(batchScan)
     )
   }
+
+  def supportsBatchScan(scan: Scan): Boolean = {
+    scan.getClass.getName == "org.apache.iceberg.spark.source.SparkBatchQueryScan"
```
[the remark this answers is not in the document]
Comment (zhztheplayer) [reply gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala:103]: `SparkBatchQueryScan` is package private. 

cc @liujiayi771 if there is some better practices for this.

File: backends-clickhouse/src/main-iceberg/scala/org/apache/gluten/component/CHIcebergComponent.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+package org.apache.gluten.component
+
+import org.apache.gluten.backendsapi.clickhouse.CHBackend
+import org.apache.gluten.execution.OffloadIcebergScan
+import org.apache.gluten.extension.injector.Injector
+
+class CHIcebergComponent extends Component {
+  override def name(): String = "clickhouse-iceberg"
+  override def buildInfo(): Component.BuildInfo =
+    Component.BuildInfo("ClickHouseIceberg", "N/A", "N/A", "N/A")
```
[the remark this answers is not in the document]
Comment (zhztheplayer) [reply backends-clickhouse/src/main-iceberg/scala/org/apache/gluten/component/CHIcebergComponent.scala:27]: Right, this part needs some cleanup. `BuildInfo` is not that general indeed. I'd like to see it totally decoupled with Component API somehow.

**Pull request merged**